apps/api/src/app/workflows-v2/util/template-parser/liquid-parser.ts

```diff
@@ -0,0 +1,126 @@
+import { Template, Liquid, RenderError, LiquidError } from 'liquidjs';
+import { TemplateParseResult, InvalidVariable } from './parser-types';
+import { isValidTemplate, extractLiquidExpressions } from './parser-utils';
+
+const LIQUID_CONFIG = {
+  strictVariables: true,
+  strictFilters: true,
+  greedy: false,
+  catchAllErrors: true,
+} as const;
+
+/**
+ * Copy of LiquidErrors type from liquidjs since it's not exported.
+ * Used to handle multiple render errors that can occur during template parsing.
+ * @see https://github.com/harttle/liquidjs/blob/d61855bf725a6deba203201357f7455f6f9b4a32/src/util/error.ts#L65
+ */
+class LiquidErrors extends LiquidError {
+  errors: RenderError[];
+}
+
+function isLiquidErrors(error: unknown): error is LiquidErrors {
+  return error instanceof LiquidError && 'errors' in error && Array.isArray((error as LiquidErrors).errors);
+}
+
+/**
+ * Parses a Liquid template string and extracts all variable names, including nested paths.
+ * Validates the syntax and separates valid variables from invalid ones.
+ *
+ * @example
+ * // Valid variables
+ * parseLiquidVariables('Hello {{user.name}}, your score is {{user.score}}')
+ * // Returns:
+ * {
+ *   validVariables: ['user.name', 'user.score'],
+ *   invalidVariables: []
+ * }
+ *
+ * @example
+ * // Mixed valid and invalid syntax
+ * parseLiquidVariables('{{user.name}} {{invalid..syntax}}')
+ * // Returns:
+ * {
+ *   validVariables: ['user.name'],
+ *   invalidVariables: [{
+ *     context: '>> 1| {{invalid..syntax}}\n                ^',
```

Comment:
Do we need the `context` key in invalidVariables?

Reply:
In my opinion, we could benefit from it. this is how the CLI presents the issues.
Let me provide an example or given example:
context: '>> 1| {{invalid..syntax}}\n                ^',

the client will have the exact place where the issue is if it is render/split by a new line
`>> 1| {{invalid..syntax}}`
`                ^`

this is an indication that Liquid JS sees an error